### ISSUE TITLE
"Azure Service Bus and more" Any reason why we subscribe to our observers (e.g. start listening to events coming in) in ConfigureServices

### ISSUE THREAD

**Issue by Hyjaz** [Microsoft/ApplicationInsights-aspnetcore#770]
## Repro Steps
1. make method call `services.AddApplicationInsightsTelemetry(someConfiguration)` inside ConfigureServices method (ASP .Net core 
2. 

##  Actual Behavior
- Subscribes observers
Here is the line we "subscribe": https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/17cc5de23ba58872650f7f2c4eaf0f0f2e47a823/src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs#L179
## Expected Behavior
- should not subscribe observers, but subscribe when we for example do a call to `UseApplicationInsights` (if exists)

## Version Info 
SDK Version : 2.3.0
.NET Version : 2.1	 						
How Application was onboarded with SDK(VisualStudio/StatusMonitor/Azure Extension) :
OS : Windows
Hosting Info (IIS/Azure WebApps/ etc) :IIS

**Comment by Hyjaz** [comment]
On AddApplicationInsights, we start listening...

I was under the impression that the pattern is that we set up our services in ConfigureServices, and then in the context of the applicationInsights start listening to service bus notification in Configure().

The issue we are having is regarding the PR you just close: https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/1038

Until we get a release with this fix, we wanted to have a workaround.
While we were trying to find a workaround, we noticed that when we do `AddApplicationInsights(configuration)`, we start listening to notification coming from Azure Service Bus and the other ones.

And it's kind of weird, to start listening on AddApplicationInsights...

so our workaround was to start listening to Azure service bus notifications before calling AddApplicationInsights, but by doing so we don't get to have the TelemetryClient through DI, and it just makes part of our structure look not clean
